src/httpserver.h

```diff
@@ -83,6 +84,19 @@ class HTTPRequest
      */
     RequestMethod GetRequestMethod() const;
 
+    /** Get the query parameter value from request uri for a specified key.
+     * Throws std::runtime_error if key does not exist and no default_val is specified.
+     *
+     * If the query string contains duplicate keys, the first value is returned. Many web frameworks
+     * would instead parse this as an array of values, but this is not (yet) implemented as it is
+     * currently not needed in any of the endpoints.
+     *
+     * @param[in] key represents the query parameter of which the value is returned
+     * @param[in] default_val (optional) is returned if key is not in query string
+     * @throws std::runtime_error if key is not in query string and no default_val was provided
```

Comment:
We don't usually use exceptions to return error conditions in the Bitcoin Core codebase. If you do this, wherever there's a key that doesn't have a default, the calling code will have to wrap this call in a try/except, and then catch the runtime_error and return it as a RESTERR.

Did you consider returning a `std::optional<std::string>`, and then using [std::optional<t>::value_or()](https://en.cppreference.com/w/cpp/utility/optional/value_or) to add the default, and [std::optional<t>::has_value()](https://en.cppreference.com/w/cpp/utility/optional/operator_bool) to catch if a non-default option hasn't been set?

Reply:
I did not consider that, but that's a great idea. I've taken it a bit further, and removed the `default_val` optional argument from GetQueryParameter, since this functionality can now just as easily be achieved in the headers endpoints by using `std::optional::value_or()` as you suggested, slightly reducing complexity of the code.